test/metabase/query_analysis/native_query_analyzer_test.clj

```diff
@@ -115,12 +139,27 @@
     (is (= {false 6}
            (frequencies (map :explicit-reference (field-refs "select v.* from venues v join checkins")))))))
 
+(deftest ^:parallel pseudo-table-fields-tests
+  (testing "When macaw returns an unknown field without a table, we keep it even if it could be phantom."
+    ;; At the time of writing this test, Macaw would (incorrectly) return "week" as an ambiguous source column.
```

Comment:
I have a hunch this will be a fairly easy fix, but things could be a bit sketchy around scope nesting.